CRM/Cdntaxreceipts/Form/ViewTaxReceipt.php

```diff
@@ -299,12 +291,18 @@ function postProcess() {
 
     $buttonName = $this->controller->getButtonName();
 
+    //CRM-1820 Once the receipt has been cancelled and user wants to preview or issue "Replace Receipt"
+    if ($this->_reissue && $this->_isCancelled && ($buttonName == '_qf_ViewTaxReceipt_submit' || $buttonName == '_qf_ViewTaxReceipt_next')){
+      list($receipt_number, $receipt_id) = CRM_Canadahelps_TaxReceipts_Receipt::receiptNumber($contribution->id,TRUE);
+      $contribution->cancelled_replace_receipt_number  = $receipt_number;
+      $contribution->replace_receipt  = 1;
+    }
+
     // If we are cancelling the tax receipt (or preview)
-    // CRM-1820 Added a check to prevent “Replace Receipt” function to enter in if condition as button name is same as preview button.
-    if ($buttonName == '_qf_ViewTaxReceipt_submit' && (!$this->_isCancelled)) {
+    if ($buttonName == '_qf_ViewTaxReceipt_submit') {
 
       // Preview
-      if (!$this->_reissue) {
+      if (!$this->_reissue || ($this->_reissue && $this->_isCancelled)) {
```

Comment:
shouldn't this simply be `!$this->_reissue || $this->_isCancelled` which translates into when cancelled or nor yet issued